wgengine/magicsock/magicsock.go

```diff
@@ -2987,6 +3089,10 @@ type discoEndpoint struct {
 	trustBestAddrUntil time.Time // time when bestAddr expires
 	sentPing           map[stun.TxID]sentPing
 	endpointState      map[netaddr.IPPort]*endpointState
+
+	// Non-nil if a outstanding "tailscale ping" command is running:
+	pendingCliPingRes  *ipnstate.PingResult
```

Comment:
Implies only one ping at a time may run. Check for that invariant in Ping() ?

Reply:
I realized with subnet routers it's not true that only one could run at a time, so made it a slice instead.